backend/app/api/endpoints/mcp_providers.py

```
@@ -0,0 +1,134 @@
+        if isinstance(prefs_data, str):
+            prefs_dict = json.loads(prefs_data)
+            return UserPreferences.model_validate(prefs_dict)
+        return UserPreferences.model_validate(prefs_data)
+    except Exception:
+        return UserPreferences()
+
+
+@router.put("/keys", response_model=MCPProviderKeysResponse)
+async def update_mcp_provider_keys(
+    keys: MCPProviderKeysRequest,
+    db: Session = Depends(get_db),
```

Ruff (0.14.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)